docs/xdsl-introduction.ipynb

```diff
@@ -0,0 +1,937 @@
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "97da7b96-b641-4915-b6b1-b38ca4cf00c5",
+   "metadata": {},
+   "source": [
+    "### Summary\n",
+    "So, we have now defined a mini-dialect using many of xDSL's features and concepts introduced earlier, including:\n",
+    "- Operations: we modelled the SQL dialect using `select` and `filter` operations\n",
+    "- Attributes: we used attributes to encode compile-time available information, from types (the `Bag` type), to constant expressions (e.g. the \"cmpi\" attribute within `arith.Cmpi`)\n",
+    "- Mixing dialects: we utilised features from other dialects (`arith`, `scf`, `builtin`)\n",
+    "- Regions and blocks: we used the operation's _regions_ to model the lambda within `sql.filter` since regions corresponds to an SCF-like representation of code. \n",
+    "    - The function arguments are modelled as the _block arguments_ of the first block in the region, and the function's code/computation is represented as the block's list of operations.\n",
+    "    - Not all operations needs a region, though. For instance `sql.select` does not have a region.\n",
+    "- xDSL type-checking: xDSL is able to use the typing information within the dialect definition to validate IRs (through `.validate()` calls)\n",
+    "- Rewrites: we defined two rewrites to transform the xDSL IR, simplifying and optimising the code (`ConstantFolding` and `DeadConstantElim`)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1faf0f6b-0f77-4130-b6ab-30977dd620fc",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f8bc26d8-d78e-4cb4-9362-7c9313a244d2",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d0a65213-e1c8-432a-b90b-00ee140efb75",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dfbdf668-3279-4092-b510-947e4ca90ce1",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true,
+    "tags": []
+   },
+   "source": [
+    "# DRAFT SECTION"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "68f04d3c-5850-4150-91ed-f3dc7f3446ee",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "# MLIR/xDSL examples\n",
+    "We want to demonstrate a range of MLIR/xDSL features\n",
+    "\n",
+    "- An SQL-like representation\n",
+    "- A subset of Python, just for demonstration purposes\n",
+    "- Toy dialect\n",
+    "- Existing dialects, e.g. scf and arith\n",
+    "\n",
+    "Would also be nice to have some examples of dialects being used together, e.g. the standard arith + if statements stuff?\n",
+    "- a range of dialect definitions with explanations, showcasing the usage/non-usage of regions, blocks, and attributes\n",
+    "- a range of MLIR/xDSL examples which makes use of the operations"
```

Comment:
I think those are good ideas!
Maybe we can open issues for them? Note that the Toy dialect is already under work :)

Reply:
sure, new issues are a good idea